geth/jail/console/console.go

```diff
@@ -0,0 +1,130 @@
+)
+
+// Extension takes a giving extension function and writer and returns a standard otto.Function
+// callable by a otto vm.
+func Extension(w io.Writer, ext func(otto.FunctionCall, io.Writer) otto.Value) func(otto.FunctionCall) otto.Value {
+	return func(fn otto.FunctionCall) otto.Value {
+		return ext(fn, w)
+	}
+}
+
+// Log provides the function caller for handling console.log
+// calls as replacement for the default console.log function within a
+// otto.Otto VM instance.
+func Log(fn otto.FunctionCall, w io.Writer) otto.Value {
+	// Record provided values into store for delviery.
+	node.SendSignal(node.SignalEnvelope{
+		Type:  EventConsoleLog,
+		Event: convertArgs(fn.ArgumentList),
+	})
+
+	// Next print out the giving values.
+	handleConsole(w, "console.log: %s", fn.ArgumentList)
+
+	return otto.UndefinedValue()
+}
+
+// Warn provides the function caller for handling console.warn
+// calls as replacement for the default console.log function within a
+// otto.Otto VM instance.
+func Warn(fn otto.FunctionCall, w io.Writer) otto.Value {
+	// Record provided values into store for delviery.
+	node.SendSignal(node.SignalEnvelope{
+		Type:  EventConsoleWarn,
+		Event: convertArgs(fn.ArgumentList),
+	})
+
+	// Next print out the giving values.
+	handleConsole(w, "console.warn: %s", fn.ArgumentList)
+
+	return otto.UndefinedValue()
+}
+
+// Debug provides the function caller for handling console.debug
+// calls as replacement for the default console.Error function within a
+// otto.Otto VM instance.
+func Debug(fn otto.FunctionCall, w io.Writer) otto.Value {
+	// Record provided values into store for delviery.
+	node.SendSignal(node.SignalEnvelope{
+		Type:  EventConsoleDebug,
+		Event: convertArgs(fn.ArgumentList),
+	})
+
+	// Next print out the giving values.
+	handleConsole(w, "console.debug: %s", fn.ArgumentList)
+
+	return otto.UndefinedValue()
+}
+
+// Error provides the function caller for handling console.error
+// calls as replacement for the default console.Error function within a
+// otto.Otto VM instance.
+func Error(fn otto.FunctionCall, w io.Writer) otto.Value {
+	// Record provided values into store for delviery.
+	node.SendSignal(node.SignalEnvelope{
+		Type:  EventConsoleError,
+		Event: convertArgs(fn.ArgumentList),
+	})
+
+	// Next print out the giving values.
+	handleConsole(w, "console.error: %s", fn.ArgumentList)
+
+	return otto.UndefinedValue()
+}
+
+// convertArgs attempts to convert otto.Values into proper go types else
+// uses original.
+func convertArgs(argumentList []otto.Value) []interface{} {
+	var items []interface{}
+
+	for _, arg := range argumentList {
+		realArg, err := arg.Export()
+		if err != nil {
+			items = append(items, arg)
+			continue
+		}
+
+		items = append(items, realArg)
+	}
+
+	return items
+}
+
+// handleConsole takes the giving otto.Values and transform as
+// needed into the appropriate writer.
+func handleConsole(writer io.Writer, format string, args []otto.Value) {
```

Comment:
Here I am also not sure if `handleConsole` is an appropriate name as it's very general. Maybe simple `writeArguments` would be more descriptive?